activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -1036,6 +1040,8 @@ def configure_connection
           add_pg_decoders
 
           reload_type_map
+
+          super
```

Comment:
Unfortunately I had to move it down because we need `@type_map_for_results` to check the version...

Reply:
Ok, I refactored the PG adapter so that `@type_map_for_result` is initialized earlier. This now allow to do the version check on an unconfigured connection.